editions/tw5.com/tiddlers/macros/TagMacro.tid

```diff
@@ -1,17 +1,59 @@
 caption: tag
 created: 20141206130540337
-modified: 20230725201240201
+modified: 20240128142041965
 tags: Macros [[Core Macros]]
 title: tag Macro
 type: text/vnd.tiddlywiki
 
+\procedure resetTagStyleTemplate()
+.tc-tag-missing {
+	font-style: initial;
+}
+\end
+
+\procedure resetTagStylesTitle() reset-tag-missing-styles
+
+\procedure resetTagFormatting()
+<% if [<resetTagStylesTitle>!has[text]] %>
+<$action-createtiddler
+	$basetitle=<<resetTagStylesTitle>>
+	text=<<resetTagStyleTemplate>>
+	code-body= "yes"
+	tags= "$:/tags/Stylesheet"
+/>
+<$action-navigate $to=<<resetTagStylesTitle>>/>
+<% else %>
+<$action-navigate $to=<<resetTagStylesTitle>>/>
+<% endif %>
+\end
+
 The <<.def tag>> [[macro|Macros]] generates a tag pill for a specified tag. Clicking the tag pill opens a dropdown. This can be compared to the [[tag-pill Macro]] which also features other parameters.
 
 <<.tip """If a [[list widget|ListWidget]] generates multiple tag macros for the same tag, clicking on one of them will open dropdowns on all of them. The simplest way to prevent this is to add the `counter="transclusion"` attribute to the list widget. See the examples below for more details.""">>
 
 !! Parameters
 
-;tag
+; tag
 : The title of the tag, defaulting to the [[current tiddler|Current Tiddler]]
 
+!! CSS classes
+
+; `tc-tag-missing`
+: This class is defined if a tag does ''not exist'' as a tiddler. By default the format is set to //italic//
+
+; `tc-tag-exists`
+: This class is defined if a tag does exist as a tiddler. This class has ''no'' default definition
+
+
+!! Disabling CSS classes
```

Comment:
I understand the intent of this section but I think it complicates the documentation. It's really a helper tool that should exist in its own right